File: multibody/test/multipose_visualizer_test.cc
#include "multibody/multipose_visualizer.h"
#include "common/find_resource.h"

int main(int argc, char* argv[]) { 
  std::srand((unsigned int) time(0));
  auto visualizer = dairlib::multibody::MultiposeVisualizer(
      dairlib::FindResourceOrThrow("examples/Cassie/urdf/cassie_v2.urdf"), 8, "pelvis");

  Eigen::MatrixXd poses = Eigen::MatrixXd::Random(30, 8);
  visualizer.DrawPoses(poses);
}

File: bindings/pydairlib/multibody/test/multipose_visualizer_test.py
from pydairlib.common import FindResourceOrThrow
from pydairlib.multibody import MultiposeVisualizer
import pydrake.systems.framework
import pydrake.multibody.plant
import pydrake.multibody.parsing
import numpy as np

# Python visualization test to draw a single state
def main():
    num_poses = 3
    alpha = np.array([.1, .3, 1])
    visualizer = MultiposeVisualizer(FindResourceOrThrow(
        "examples/Cassie/urdf/cassie_v2.urdf"), num_poses, alpha, "")
    # Create a plant just to know number of positions (also as a pydrake test)
    builder = pydrake.systems.framework.DiagramBuilder()
    plant, scene_graph = pydrake.multibody.plant.AddMultibodyPlantSceneGraph(
        builder, 0)
    pydrake.multibody.parsing.Parser(plant).AddModelFromFile(
        FindResourceOrThrow("examples/Cassie/urdf/cassie_v2.urdf"))
    plant.Finalize()
    visualizer.DrawPoses(np.random.rand(plant.num_positions(),
        num_poses))

if __name__ == "__main__":
    main()
